catroid/src/org/catrobat/catroid/stage/ShowTextActor.java

```diff
@@ -59,15 +61,28 @@ public ShowTextActor(String text, int x, int y) {
 	@Override
 	public void draw(Batch batch, float parentAlpha) {
 		DataContainer projectVariableContainer = ProjectManager.getInstance().getCurrentProject().getDataContainer();
-		List<UserVariable> variableList = projectVariableContainer.getProjectVariables();
+		List<UserVariable> projectVariableList = projectVariableContainer.getProjectVariables();
+
+		Map<Sprite, List<UserVariable>> spriteVariableMap = projectVariableContainer.getSpriteVariableMap();
+		Sprite currentSprite = ProjectManager.getInstance().getCurrentSprite();
+		List<UserVariable> spriteVariableList = spriteVariableMap.get(currentSprite);
 
 		if (variableName.equals("No variable set")) {
```

Comment:
Not in the current changeset but still relevant. Besides it's a hardcoded string used in multiple locations, what would happen if someone uses "No variable set" as variable name:question: